6.5.group_by.py

```diff
@@ -0,0 +1,13 @@
+def group_by(func, L):
+    d = {}
+    for i in L:
+        x = func(i)
+        if x in d:
+            d[x].append(i)
+        else:
+            d[x] = [i]
+    return d
```

Comment:
replace 'd' 'i' 'l' 'x' these are not good varible names Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

 You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.